src/python/zquantum/qaoa/estimators.py

```diff
@@ -45,35 +44,49 @@ def get_estimated_expectation_values(
         Returns:
             ExpectationValues: expectation values for each term in the target operator.
         """
-        if alpha > 1 or alpha <= 0:
+        if self.alpha > 1 or self.alpha <= 0:
             raise ValueError("alpha needs to be a value between 0 and 1.")
```

Comment:
This is totally fine as is, but usage would probably be better if `alpha` was a property and this could be raised during the `setter`

Reply:
I would see how we deal with other, more complicated estimators and then fix this one accordingly for the consistency sake.